src/alerts/version-alerts.ts

```diff
@@ -54,3 +55,34 @@ export const getBadVersions = async (): Promise<string[]> => {
     throw error;
   }
 };
+
+export const useTimeUntilRequiredUpdate = (): number | undefined => {
+  const { updateRequiredBy } = useStore(versionStore);
+
+  const [timeRemaining, setTimeRemaining] = useState(
+    updateRequiredBy ? Math.floor((updateRequiredBy - Date.now()) / 1000) : undefined
+  );
+  const updateTimeRemaining = useCallback(() => {
+    if (updateRequiredBy) {
+      const timeRemaining = Math.floor((updateRequiredBy - Date.now()) / 1000);
+      if (timeRemaining <= 0) {
+        window.location.reload();
```

Comment:
How does `updateRequiredBy` then get cleared (so that we don't keep updating)? Is it because the new version updates the `versionStore` and it clears out `updateRequiredBy`?

Reply:
When the page reloads, all the app's state is reinitialized.